File: frontend/src/api/wiki.ts
import apiClient from './client'

export interface WikiPage {
  id: string
  title: string
  importance: 'high' | 'medium' | 'low'
  content_md: string
  relevant_files: string[]
  order_index: number
}

export interface WikiSection {
  id: string
  title: string
  order_index: number
  pages: WikiPage[]
}

export interface WikiResponse {
  id: string
  repo_id: string
  title: string
  llm_provider: string
  llm_model: string
  created_at: string
  sections: WikiSection[]
}

export interface RegenerateWikiRequest {
  llm_provider?: string
  llm_model?: string
  pages?: string[]
}

export interface RegenerateWikiResponse {
  task_id: string
  message: string
}

export async function getWiki(repoId: string): Promise<WikiResponse> {
  const response = await apiClient.get<WikiResponse>(`/wiki/${repoId}`)
  return response.data
}

export async function regenerateWiki(
  repoId: string,
  data?: RegenerateWikiRequest
): Promise<RegenerateWikiResponse> {
  const response = await apiClient.post<RegenerateWikiResponse>(
    `/wiki/${repoId}/regenerate`,
    data || {}
  )
  return response.data
}

export async function deleteWiki(repoId: string): Promise<void> {
  await apiClient.delete(`/wiki/${repoId}`)
}

export async function getWikiPage(repoId: string, pageId: string): Promise<WikiPage> {
  const response = await apiClient.get<WikiPage>(`/wiki/${repoId}/pages/${pageId}`)
  return response.data
}

File: app/schemas/wiki.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class WikiPageResponse(BaseModel):
    id: str
    title: str
    importance: str
    content_md: Optional[str] = None
    relevant_files: Optional[List[str]] = None
    order_index: int

    model_config = {"from_attributes": True}


class WikiSectionResponse(BaseModel):
    id: str
    title: str
    order_index: int
    pages: List[WikiPageResponse] = []

    model_config = {"from_attributes": True}


class WikiResponse(BaseModel):
    id: str
    repo_id: str
    title: str
    llm_provider: Optional[str] = None
    llm_model: Optional[str] = None
    created_at: datetime
    sections: List[WikiSectionResponse] = []

    model_config = {"from_attributes": True}


class WikiRegenerateRequest(BaseModel):
    llm_provider: Optional[str] = None
    llm_model: Optional[str] = None
    pages: Optional[List[str]] = None  # 可选：仅重新生成指定页面 ID 列表


class WikiRegenerateResponse(BaseModel):
    task_id: str
    message: str = "Wiki 重新生成任务已提交"
